appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -35,9 +40,29 @@ protected MockChart(SimpleEditor editor, String type, ImageResource icon) {
     protected void initChart() {
         chartWidget.getOptions().setMaintainAspectRatio(false);
         chartWidget.getOptions().getTitle().setDisplay(true);
-        chartWidget.setStylePrimaryName("ode-SimpleMockComponent");
 
-        initComponent(chartWidget);
+        // Since the Mcok Chart component is not a container in a normal
+        // sense (attached components should not be visible), the Chart Widget
+        // is added to the root panel, and the root panel itself is initialized.
+        // This is done to ensure that Mock Chart Data components can be dragged
+        // onto the Chart itself, rather than outside the Chart component.
+        rootPanel.setStylePrimaryName("ode-SimpleMockComponent");
+        rootPanel.add(chartWidget);
```

Comment:
Is it okay to add the Chart Widget to the Root Panel?

This seemed like a good approach since the only visible component in this panel will be the Chart itself, and adding the widget to the root panel allows Data components to be added as they are dragged on the Chart itself.

This seemed to work in practice really well.

Reply:
Also, is it okay to initialize the rootPanel, rather than another GWT component? (see line 53)